File: project_euler/00/005.go
// Copyright [2017] <dmnsn7@gmail.com>

package main

import "fmt"

const n = 20

func gcd(a, b int64) int64 {
	if b == 0 {
		return a
	}
	return gcd(b, a%b)
}

func lcm(a, b int64) int64 {
	return a / gcd(a, b) * b
}

func main() {
	ans := int64(1)
	for i := 1; i <= n; i++ {
		ans = lcm(ans, int64(i))
	}

	fmt.Printf("%d\n", ans)
}

File: project_euler/00/005.py
#!/usr/bin/env python3
'''Copyright [2017] <dmnsn7@gmail.com>'''

N = 20


def gcd(num_a, num_b):
    return gcd(num_b, num_a % num_b) if num_b else num_a


def lcm(num_a, num_b):
    return num_a // gcd(num_a, num_b) * num_b


def main():
    ans = 1
    for i in range(1, N + 1):
        ans = lcm(ans, i)

    print(ans)


if __name__ == '__main__':
    main()
